mm2src/coins/eth/eth_balance_events.rs

```diff
@@ -1,32 +1,61 @@
+use super::EthCoin;
+use crate::{eth::{u256_to_big_decimal, Erc20TokenInfo},
+            BalanceError, CoinWithDerivationMethod};
+use common::{executor::Timer, log, Future01CompatExt};
+use mm2_event_stream::{Broadcaster, Event, EventStreamer, NoDataIn, StreamHandlerInput};
+use mm2_number::BigDecimal;
+
 use async_trait::async_trait;
-use common::{executor::{AbortSettings, SpawnAbortable, Timer},
-             log, Future01CompatExt};
 use ethereum_types::Address;
-use futures::{channel::oneshot::{self, Receiver, Sender},
-              stream::FuturesUnordered,
-              StreamExt};
+use futures::{channel::oneshot, stream::FuturesUnordered, StreamExt};
 use instant::Instant;
-use mm2_core::mm_ctx::MmArc;
-use mm2_err_handle::prelude::MmError;
-use mm2_event_stream::{behaviour::{EventBehaviour, EventInitStatus},
-                       ErrorEventName, Event, EventName, EventStreamConfiguration};
-use mm2_number::BigDecimal;
+use serde::Deserialize;
+use serde_json::Value as Json;
 use std::collections::{HashMap, HashSet};
 
-use super::EthCoin;
-use crate::{eth::{u256_to_big_decimal, Erc20TokenInfo},
-            BalanceError, CoinWithDerivationMethod, MmCoin};
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields, default)]
+struct EthBalanceStreamingConfig {
+    /// The time in seconds to wait before re-polling the balance and streaming.
+    pub stream_interval_seconds: f64,
+}
+
+impl Default for EthBalanceStreamingConfig {
+    fn default() -> Self {
+        Self {
+            stream_interval_seconds: 10.0,
+        }
+    }
+}
+
+pub struct EthBalanceEventStreamer {
+    /// The period in seconds between each balance check.
+    interval: f64,
+    coin: EthCoin,
+}
```

Comment:
I think `stream_interval_seconds` should be optional. You implemented Default implementation for it below ?

Reply:
yeah we get the default if it's missing. doesn't have to be optional.